dom/src/main/scala/Dom.scala

```diff
@@ -33,7 +34,9 @@ object Dom {
     dom.asInstanceOf[Async[F]]
 }
 
-trait Node[F[_]] extends js.Any
+@JSGlobal
+@js.native
+class Node[F[_]] extends js.Any
```

Comment:
We can keep the constructor private.

```suggestion
class Node[F[_]] private () extends js.Any
```

Reply:
Gah! I knew I'd forgotten something, lemme rerun the macro.

Reply:
So I _think_ it needs to be `protected` to allow anything to extend it. We could `seal` it instead, but it would mean that writing extensions elsewhere is hard to do? wdyt?

Reply:
Oh yeah, `protected` is probably fine 🤔 basically we want to discourage users from calling `new` too easily.